src/test/java/org/mockitousage/annotation/SpyAnnotationTest.java

```diff
@@ -173,18 +192,86 @@ public void should_report_when_encosing_instance_is_needed() throws Exception {
         }
     }
 
-    static class NestedClassWithoutDefinedConstructor { }
+    @Test
+    public void should_report_private_inner_not_supported() throws Exception {
+        try {
+            MockitoAnnotations.initMocks(new WithInnerPrivate());
+            fail();
+        } catch (MockitoException e) {
+            // Currently fails at instantiation time, because the mock subclass don't have the
+            // 1-arg constructor expected for the outerclass.
+            // org.mockito.internal.creation.instance.ConstructorInstantiator.withParams()
+            assertThat(e).hasMessageContaining("Unable to initialize @Spy annotated field 'spy_field'")
```

Comment:
Ah right of course. However though, one can write : `private class InnerPrivateSub extends InnerPrivate {}` in the same enclosing class. Then this `InnerPrivateSub` will be instantiable, and will have a single private 1-arg (of the enclosing class type) constructor.

Here's bytecode outline

```
// class version 50.0 (50)
// access flags 0x20
class org/mockitousage/annotation/SpyAnnotationTest$WithInnerPrivate$InnerPrivateSub extends org/mockitousage/annotation/SpyAnnotationTest$WithInnerPrivate$InnerPrivate  {

  // compiled from: SpyAnnotationTest.java
  // access flags 0x8
  static INNERCLASS org/mockitousage/annotation/SpyAnnotationTest$WithInnerPrivate org/mockitousage/annotation/SpyAnnotationTest WithInnerPrivate
  // access flags 0x2
  private INNERCLASS org/mockitousage/annotation/SpyAnnotationTest$WithInnerPrivate$InnerPrivateSub org/mockitousage/annotation/SpyAnnotationTest$WithInnerPrivate InnerPrivateSub
  // access flags 0x2
  private INNERCLASS org/mockitousage/annotation/SpyAnnotationTest$WithInnerPrivate$InnerPrivate org/mockitousage/annotation/SpyAnnotationTest$WithInnerPrivate InnerPrivate
  // access flags 0x1008
  static synthetic INNERCLASS org/mockitousage/annotation/SpyAnnotationTest$1 org/mockitousage/annotation/SpyAnnotationTest null

  // access flags 0x1010
  final synthetic Lorg/mockitousage/annotation/SpyAnnotationTest$WithInnerPrivate; this$0

  // access flags 0x2
  private <init>(Lorg/mockitousage/annotation/SpyAnnotationTest$WithInnerPrivate;)V
   L0
    LINENUMBER 263 L0
    ALOAD 0
    ALOAD 1
    PUTFIELD org/mockitousage/annotation/SpyAnnotationTest$WithInnerPrivate$InnerPrivateSub.this$0 : Lorg/mockitousage/annotation/SpyAnnotationTest$WithInnerPrivate;
    ALOAD 0
    ALOAD 1
    ACONST_NULL
    INVOKESPECIAL org/mockitousage/annotation/SpyAnnotationTest$WithInnerPrivate$InnerPrivate.<init> (Lorg/mockitousage/annotation/SpyAnnotationTest$WithInnerPrivate;Lorg/mockitousage/annotation/SpyAnnotationTest$1;)V
    RETURN
   L1
    LOCALVARIABLE this Lorg/mockitousage/annotation/SpyAnnotationTest$WithInnerPrivate$InnerPrivateSub; L0 L1 0
    MAXSTACK = 3
    MAXLOCALS = 2
}
```

I think the `INNERCLASS` represents the `InnerClasses` attributes as specified in [§4.7.5](https://docs.oracle.com/javase/specs/jvms/se6/html/ClassFile.doc.html#79996).